service/matching/matchingEngine.go

```diff
@@ -242,6 +243,20 @@ pollLoop:
 			return nil, err
 		}
 
+		if tCtx.queryTaskInfo != nil {
+			// for query task, we don't need to update history to record decision task started, so we would use a fake
+			// RecordDecisionTaskStartedResponse with large startEventID so when frontend receives the response, it will
+			// load the entire history (eventID < StartedEventID + 1). We also use the same large number as
```

Comment:
It is dangerous to use load all the history on this fake startedEventID.  Append only model could leave some zombie events towards the end of history.  We still need to make sure we don't load events beyond NextEventID.

Reply:
then we would have to call GetWorkflowExecutionNextEventID() to get the NextEventID.